lib/ExpensiMark.ts

```diff
@@ -115,6 +176,9 @@ export default class ExpensiMark {
                 name: 'heading1',
                 process: (textToProcess, replacement, shouldKeepRawInput = false) => {
                     const regexp = shouldKeepRawInput ? /^# ( *(?! )(?:(?!<pre>|\n|\r\n).)+)/gm : /^# +(?! )((?:(?!<pre>|\n|\r\n).)+)/gm;
+                    if (typeof replacement === 'function') {
+                        return textToProcess.replace(regexp, (...args) => replacement(EXTRAS_DEFAULT, ...args));
+                    }
                     return textToProcess.replace(regexp, replacement);
```

Comment:
I see that in 5 places we are doing something similar. What about moving this logic to separate function?
For example:

```ts
function replaceText(text: string, regexp: RegExp, extras, replacement: Replacement) {
    if (typeof replacement === 'function') {
        return text.replace(regexp, (...args) => replacement(extras, ...args));
    }
    return text.replace(regexp, replacement);
}
```